src/psyclone/tests/psyir/symbols/symbol_table_test.py

```diff
@@ -2482,6 +2484,110 @@ def test_resolve_imports_parent_scope(fortran_reader, tmpdir, monkeypatch):
     assert new_sym.interface.container_symbol.name == "a_mod"
 
 
+def test_resolve_imports_from_child_symtab(
+        fortran_reader, tmpdir, monkeypatch):
+    '''Check that when an unresolved symbol is declared in a subroutine,
+    resolve imports can resolve it from a parent module as long as
+    there are no wildcard imports in the subroutine.
+
+    '''
+    # Set up include_path to import the proper modules
+    monkeypatch.setattr(Config.get(), '_include_paths', [str(tmpdir)])
+    filename = os.path.join(str(tmpdir), "a_mod.f90")
+    with open(filename, "w", encoding='UTF-8') as module:
+        module.write('''
+        module a_mod
+            integer :: some_var
+        end module a_mod
+        ''')
+    psyir = fortran_reader.psyir_from_source('''
+        module b_mod
+            use a_mod
+        contains
+          subroutine my_sub()
+            some_var = 0.0
+          end subroutine
+        end module b_mod
+        ''')
```

Comment:
The remaining NEMO issues could be from the case:
```
    with open(filename, "w", encoding='UTF-8') as module:
        module.write('''
        module a_mod
            integer :: some_var
        end module a_mod
        ''')
    psyir = fortran_reader.psyir_from_source('''
        module b_mod
            use a_mod
        contains
          subroutine my_sub()
            some_var = 0.0
          end subroutine
          subroutine my_sub2()
            some_var = 0.0
          end subroutine
        end module b_mod
        ''')
```
Any of the two subroutines have imports or unknwonfortrantypes, can you see if for such a case we can resolve the two generic same_var symbols into a single module datasymbol?

Reply:
This now seems to work OK.